openwisp-config/files/openwisp.init

```diff
@@ -1,119 +1,86 @@
 		# replace spaces with commas to avoid problems when
 		# passing this arg to procd_set_param command
 		unmanaged=$(echo $unmanaged | tr ' ' ',')
 		unmanaged="--unmanaged $unmanaged";
 	fi
-	if [ $merge_config ]; then merge_config="--merge-config $merge_config"; fi
-	if [ $test_config ]; then test_config="--test-config $test_config"; fi
-	if [ $test_script ]; then test_script="--test-script $test_script"; fi
-	if [ $connect_timeout ]; then connect_timeout="--connect-timeout $connect_timeout"; fi
-	if [ $max_time ]; then max_time="--max-time $max_time"; fi
-	if [ $capath ]; then capath="--capath $capath"; fi
-	if [ $cacert ]; then cacert="--cacert $cacert"; fi
-	if [ $mac_interface ]; then mac_interface="--mac-interface $mac_interface"; fi
-	if [ $management_interface ]; then management_interface="--management-interface $management_interface"; fi
-	if [ $pre_reload_hook ]; then pre_reload_hook="--pre-reload-hook $pre_reload_hook"; fi
-	if [ $post_reload_hook ]; then post_reload_hook="--post-reload-hook $post_reload_hook"; fi
-	if [ $post_registration_hook ]; then post_registration_hook="--post-registration-hook $post_registration_hook"; fi
 
-	if [ -z "$url" ]; then
-		logger -s "url is not set, please add it in /etc/config/openwisp" \
-		       -t openwisp \
-		       -p daemon.err
-		exit 1
-	fi
+	procd_open_instance
+	procd_set_param command $PROG $unmanaged
 
-	if ([ -z "$uuid" ] || [ -z "$key" ]) && [ -z "$shared_secret" ]; then
-		logger -s "you must either set uuid and key, or shared_secret in /etc/config/openwisp" \
-		       -t openwisp \
-		       -p daemon.err
-		exit 1
-	fi
+	config_foreach parse_config controller
 
-	procd_open_instance
-	procd_set_param command $PROG $url $interval $verify_ssl $uuid $key $shared_secret \
-	                        $consistent_key $hardware_id_script $hardware_id_key \
-	                        $bootup_delay $unmanaged $merge_config $test_config \
-	                        $test_script $connect_timeout $max_time $capath $cacert \
-	                        $mac_interface $management_interface $pre_reload_hook \
-	                        $post_reload_hook $post_registration_hook
 	procd_set_param respawn
 	procd_close_instance
-	logger -s "$PROG_NAME started" \
-	       -t openwisp \
-	       -p daemon.info
 }
 
 service_triggers() {
 	procd_add_reload_trigger openwisp
 }
 
-stop_service() {
-	logger -s "$PROG_NAME stopping" \
-	       -t openwisp \
-	       -p daemon.info
-}
-
 reload_service() {
-	logger -s "$PROG_NAME received reload trigger" \
-		   -t openwisp \
-		   -p daemon.info
-	local control_file=/tmp/openwisp/applying_conf
 	# avoid reloading while configuration is being applied
 	# will wait for a maximum of 30 seconds
 	for i in $(seq 1 30); do
-		if [ -f $control_file ]; then
+		if [ -f "$CONTROL_FILE" ]; then
 			sleep 1
 		else
 			break
 		fi
 	done
-	# ensure control file is removed
-	rm $control_file
-	# reload
-    start
+	rm "$CONTROL_FILE"
+
+	start
```

Comment:
shouldn't it be 

```
stop
start
```
?

Reply:
I'll have to look into that.
Not sure if start triggers an implicit stop.
Wouldn't change a thing now.

Reply:
It doesn't as I tested it. By reading the openwrt docs, another options seems to be capturing the SIGHUP signal in the agent.

Let's leave it by now, I'll open a new issue to not forget about it

Reply:
I have checked it again, and it works fine as long as you have updated your configuration file, so we can ignore it :)